crates/oxc_transformer/src/jsx/refresh.rs

```diff
@@ -32,20 +32,20 @@ enum RefreshIdentifierResolver<'a> {
 impl<'a> RefreshIdentifierResolver<'a> {
     /// Parses a string into a RefreshIdentifierResolver
     pub fn parse(input: &str, ast: AstBuilder<'a>) -> Self {
-        if !input.contains('.') {
-            // Handle simple identifier reference
-            return Self::Identifier(ast.identifier_reference(SPAN, input));
-        }
-
         let mut parts = input.split('.');
+
         let first_part = parts.next().unwrap();
+        let Some(second_part) = parts.next() else {
+            // Handle simple identifier reference
+            return Self::Identifier(ast.identifier_reference(SPAN, input));
+        };
```

Comment:
Just curious, is there a way to get the second part first?

Reply:
I don't think so. `Split` is an iterator, so you have to consume it in order. `first_part` is just a slice of `input` so it's not costly to generate.

I would also *hope* that compiler knows that `Split` always yields at least 1 item (even `"".split('.')` yields a single item `""`), so it should be able to remove the `.unwrap()` in `let first_part = parts.next().unwrap();`. I don't know that for sure, though.